### ISSUE TITLE
Is this repo unmaintained?

### ISSUE THREAD

**Issue by sompylasar** [FormidableLabs/nuka-carousel#213]

**Comment by kenwheeler** [comment]
I'm planning on doing a bit of a rewrite in the very near future. This was written when we still used createClass, so I should be able to vastly improve performance and stability, while keeping the existing API.

**Comment by sompylasar** [comment]
@kenwheeler Maintaining an open-source repo is replying to pull requests (30 hanging as of now) and issues (66 as of now) before doing any major rewrite that will lead to even more issues and PRs. Please address these first. Thank you!

**Comment by kenwheeler** [comment]
@sompylasar I think I'm going to go ahead and personally make the call on priority for my repo. Restructuring away from createClass is priority in my opinion, as createClass is now deprecated.

**Comment by nnjpp** [comment]
Thanks for all your work.

If anyone is interested:

https://github.com/FormidableLabs/nuka-carousel/compare/master...petflow:master#diff-3b3282d4ffc50855fd21a0660d26ff97

I spend a good chunk of the day moving away from createClass for my own purposes (didn't know you were planning on doing the same) and it's running smoothly so far. It's quick and dirty - needs refactoring in certain areas like the ugly binds in constructor (arrow fns ? but this adds a hard dependency on babel stage 2) 

The bright side is that it runs without any console warnings/errors on latest React.

**Comment by kenwheeler** [comment]
@nnjpp thats awesome. I'm trying to figure out a way to get rid of tweenState right now, have it mostly refactored into es2016-ish code. Do you guys have any ideas? I can totally use react-mixin, but I feel like if I either move it to motion or a custom physics based animator, i could add free scroll mode, which is hotly requested.

**Comment by sompylasar** [comment]
@kenwheeler This may help, it's not a mixin but an aggregate: https://github.com/tejitak/react-state-animation

**Comment by sompylasar** [comment]
@nnjpp FYI https://medium.com/@boennemann/avoid-yarn-for-packages-and-fully-enjoy-its-benefits-for-application-development-8bdd4deb33cf

**Comment by nnjpp** [comment]
@sompylasar thanks, good to know. I've been tentatively using it for app development but i'll take it out of this fork

**Comment by alvaropinot** [comment]
@nnjpp take care with those binds as with the arrow functions because the can lead to some perf issues. I've found this article really interesting into that kind of anti-patterns 👇 
 https://twitter.com/AlvaroPinot/status/872333256221478916. 

Hope it helps :) 

BTW @kenwheeler this reminds me that I've observed multiple calls to the render function, probably because of this kind of issues. Have you checked that in your latest rewrite? :)
